hackathon_site/dashboard/frontend/src/components/general/Header/Header.js

```diff
@@ -5,39 +5,71 @@ import AppBar from "@material-ui/core/AppBar";
 import Toolbar from "@material-ui/core/Toolbar";
 import Typography from "@material-ui/core/Typography";
 import IconButton from "@material-ui/core/IconButton";
+import Drawer from "@material-ui/core/Drawer";
 import Hidden from "@material-ui/core/Hidden";
 import Menu from "@material-ui/icons/Menu";
-import Navbar from "components/general/Navbar/Navbar";
+import { Navbar, NavbarDrawer } from "components/general/Navbar/Navbar";
+import { userEmail, cartQuantity } from "testing/mockData";
 
-const Header = ({ showNavbar = true }) => (
-    <AppBar className={styles.header} position="sticky">
-        <Toolbar className={styles.headerToolbar}>
-            <Hidden implementation="css" smDown={!!showNavbar}>
-                <div className={styles.headerLogo}>
-                    <Logo
-                        className={styles.headerLogoImg}
-                        alt="Hackathon logo"
-                        data-testid="headerLogo"
-                    />
-                    <Typography variant="h6" data-testid="hackathonName">
-                        Hackathon Name
-                    </Typography>
-                </div>
-            </Hidden>
-            <Hidden implementation="css" mdUp>
-                {showNavbar && (
-                    <IconButton
-                        color="inherit"
-                        aria-label="Menu"
-                        className={styles.headerIconBtn}
-                    >
-                        <Menu />
-                    </IconButton>
-                )}
-            </Hidden>
-            {showNavbar && <Navbar />}
-        </Toolbar>
-    </AppBar>
+const HackathonTitle = () => (
+    <>
+        <Logo
+            className={styles.headerLogoImg}
+            alt="Hackathon logo"
+            data-testid="headerLogo"
+        />
+        <Typography variant="h6" data-testid="hackathonName">
+            Hackathon Name
+        </Typography>
+    </>
 );
 
+const Header = ({ showNavbar = true }) => {
+    const [mobileOpen, setMobileOpen] = React.useState(false);
+
+    const toggleMenu = () => {
+        setMobileOpen(!mobileOpen);
+    };
+
+    return (
+        <AppBar className={styles.header} position="sticky">
+            <Toolbar className={styles.headerToolbar}>
+                <Hidden implementation="css" smDown={!!showNavbar}>
```

Comment:
Functionality wise, I have no issues setting this to the JS implementation when I tested. I'd rather we do that, since it reduces how much we have to render (which I think offsets the extra JS processing to check the page width).